File: SDKs/clickmeter.com/v2/python/src/sdk/models/operations/account_getpermissionscount.py
from dataclasses import dataclass, field
from typing import Optional
from enum import Enum
from sdk.models import shared


@dataclass
class AccountGetPermissionsCountPathParams:
    guest_id: int = field(metadata={'path_param': { 'field_name': 'guestId', 'style': 'simple', 'explode': False }})
    
class AccountGetPermissionsCountEntityTypeEnum(str, Enum):
    DATAPOINT = "datapoint"
    GROUP = "group"

class AccountGetPermissionsCountTypeEnum(str, Enum):
    R = "r"
    W = "w"


@dataclass
class AccountGetPermissionsCountQueryParams:
    entity_id: Optional[int] = field(default=None, metadata={'query_param': { 'field_name': 'entityId', 'style': 'form', 'explode': True }})
    entity_type: Optional[AccountGetPermissionsCountEntityTypeEnum] = field(default=None, metadata={'query_param': { 'field_name': 'entityType', 'style': 'form', 'explode': True }})
    type: Optional[AccountGetPermissionsCountTypeEnum] = field(default=None, metadata={'query_param': { 'field_name': 'type', 'style': 'form', 'explode': True }})
    

@dataclass
class AccountGetPermissionsCountRequest:
    path_params: AccountGetPermissionsCountPathParams = field()
    query_params: AccountGetPermissionsCountQueryParams = field()
    

@dataclass
class AccountGetPermissionsCountResponse:
    content_type: str = field()
    status_code: int = field()
    api_core_responses_count_responce: Optional[shared.APICoreResponsesCountResponce] = field(default=None)
    body: Optional[bytes] = field(default=None)
    

File: SDKs/clickmeter.com/v2/typescript/src/sdk/models/operations/accountgetpermissionscount.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class AccountGetPermissionsCountPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=guestId" })
  guestId: number;
}

export enum AccountGetPermissionsCountEntityTypeEnum {
    Datapoint = "datapoint",
    Group = "group"
}

export enum AccountGetPermissionsCountTypeEnum {
    R = "r",
    W = "w"
}


export class AccountGetPermissionsCountQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=entityId" })
  entityId?: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=entityType" })
  entityType?: AccountGetPermissionsCountEntityTypeEnum;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=type" })
  type?: AccountGetPermissionsCountTypeEnum;
}


export class AccountGetPermissionsCountRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: AccountGetPermissionsCountPathParams;

  @SpeakeasyMetadata()
  queryParams: AccountGetPermissionsCountQueryParams;
}


export class AccountGetPermissionsCountResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  apiCoreResponsesCountResponce?: shared.ApiCoreResponsesCountResponce;

  @SpeakeasyMetadata()
  body?: Uint8Array;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}
